src/sfdc/salesforceBaseUtils/salesforce_errors.bal

```diff
@@ -16,30 +16,31 @@
 // under the License.
 //
 
-import ballerina/lang.'error as errs;
 
-# Holds the details of an Salesforce error
-# + errorCode - Error code for the error
+# Holds the details of an Salesforce error.
+#
+# + errorCode - Error code for the error 
+# + cause - Root error that caused the Salesforce error
 type ErrorDetail record {
-    *errs:Detail;
     string errorCode;
+    error cause?;
```

Comment:
The `cause` is the second param of error construct. So defining it in the `ErrorDetail` seems unnecessary. Remove the documentation as well